velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -1277,6 +1277,72 @@ TEST_F(DateTimeFunctionsTest, millisecondTimestampWithTimezone) {
           "millisecond(c0)", -980, "+05:30"));
 }
 
+TEST_F(DateTimeFunctionsTest, lastDayOfMonth) {
+  const auto lastDayOfMonth = [&](std::optional<Timestamp> timestamp) {
+    return evaluateOnce<Date>("last_day_of_month(c0)", timestamp);
+  };
+  EXPECT_EQ(std::nullopt, lastDayOfMonth(std::nullopt));
+  EXPECT_EQ(Date(30), lastDayOfMonth(Timestamp(0, 0)));
+  EXPECT_EQ(Date(15370), lastDayOfMonth(Timestamp(1325376000, 0)));
+  EXPECT_EQ(Date(15370), lastDayOfMonth(Timestamp(1328054399, 0)));
+
+  setQueryTimeZone("Pacific/Apia");
+
+  EXPECT_EQ(std::nullopt, lastDayOfMonth(std::nullopt));
+  EXPECT_EQ(Date(-1), lastDayOfMonth(Timestamp(0, 0)));
+  EXPECT_EQ(Date(15370), lastDayOfMonth(Timestamp(1325376000, 0)));
+  EXPECT_EQ(Date(15399), lastDayOfMonth(Timestamp(1328054399, 0)));
+}
+
+TEST_F(DateTimeFunctionsTest, lastDayOfMonthDate) {
+  const auto lastDayOfMonth = [&](std::optional<Date> date) {
+    return evaluateOnce<Date>("last_day_of_month(c0)", date);
+  };
+  EXPECT_EQ(std::nullopt, lastDayOfMonth(std::nullopt));
+  EXPECT_EQ(Date(19388), lastDayOfMonth(Date(19358))); // 2022-01-01
```

Comment:
This is a bit hard to read.

Perhaps, modify lastDayOfMonth to take date as a string and return a string:

```
EXPECT_EQ("2022-01-31", lastDayOfMonth("2022-01-01"));
```